src/Microsoft.Dnx.TestHost/Project.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+
+namespace Microsoft.Dnx.TestHost
+{
+    public class Project
+    {
+        private const string ProjectFileName = "project.json";
+
+        public string Name { get; set; }
+
+        public IDictionary<string, string> Commands { get; }
+            = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+        public static bool TryGetProject(string inputPath, out Project project)
+        {
+            project = null;
+
+            string projectDirPath;
+            if (string.Equals(Path.GetFileName(inputPath), ProjectFileName, StringComparison.OrdinalIgnoreCase))
+            {
+                projectDirPath = Path.GetDirectoryName(inputPath);
+            }
+            else
+            {
+                projectDirPath = inputPath;
+            }
+
+            // Assume the directory name is the project name if none was specified
+            var projectName = GetDirectoryName(projectDirPath);
+            var projectJsonPath = Path.Combine(projectDirPath, ProjectFileName);
+            projectJsonPath = Path.GetFullPath(projectJsonPath);
+
+            if (!File.Exists(projectJsonPath))
+            {
+                return false;
+            }
+
+            project = GetProject(File.OpenRead(projectJsonPath), projectJsonPath, projectName);
```

Comment:
I am doing a `CloseInput = true` on the Json text reader code below this method.